lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -274,8 +274,8 @@ function _campaign_resource_signup($nid, $values) {
     unset($values['northstar_id']);
   }
 
-  // If we pass either a Northstar ID or Drupal UID to `uid`, convert it to a real UID.
-  $values['uid'] = dosomething_user_convert_to_legacy_id($values['uid']);
+  // If a Drupal UID is passed to `uid`, convert it to the user's Northstar ID.
+  $values['uid'] = dosomething_user_convert_to_northstar_id($values['uid']);
```

Comment:
ah actually I'm getting a little mixed up here @DFurnes - sorry about this! So unless I'm understanding incorrectly, I don't think we can just bypass any code that relies on a `uid` or loaded `$user` object if a Northstar ID is provided because of functions like [`dosomething_third_party_subscribe`](https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_signup/dosomething_signup.module#L293). This function uses others to find out user information (age, language, country)... unless it is ok to bypass this as well? 

Is the idea here to just make sure the user's signup and reportback gets logged in Rogue?